pkg/workload/tpch/tpch.go

```diff
@@ -283,9 +295,21 @@ func (w *worker) run(ctx context.Context) error {
 		}
 		numRows++
 	}
+
+	// It is possible that we short-circuited the loop above, and we need to
+	// fully consume the rows before we might get an error from the server.
+	for rows.Next() {
```

Comment:
I think a rows.Close() is a better choice here.

Reply:
Yeah I think so too. You don't need to consume the whole thing to get its error I don't believe.

Also, hmm, is it expected from Postgres that you might have to consume the whole result set before seeing an error? I think probably not, which makes me a little bit nervous. If someone read 3 rows, decided to stop reading, but we had this weird error condition, they would have gotten incorrect results... not something to fix here but we should probably investigate this a little more.